website/content/v1.4/kubernetes-guides/network/deploying-cilium.md

```diff
@@ -5,44 +5,99 @@ aliases:
-Adhering to Talos principles we'll deploy Cilium with IPAM mode set to Kubernetes.
-Each method can either install Cilium using kube proxy (default) or without: [Kubernetes Without kube-proxy](https://docs.cilium.io/en/v1.11/gettingstarted/kubeproxy-free/)
+This documentation will outline installing Cilium CNI v1.13.0 on Talos in six different ways.
+Adhering to Talos principles we'll deploy Cilium with IPAM mode set to Kubernetes, and using the `cgroupv2` mount that talos already provides.
+As Talos does not allow loading kernel modules by Kubernetes workloads, `SYS_MODULE` capability needs to be dropped from the Cilium default set of values, this override can be seen in the helm/cilium cli install commands.
+Each method can either install Cilium using kube proxy (default) or without: [Kubernetes Without kube-proxy](https://docs.cilium.io/en/v1.13/network/kubernetes/kubeproxy-free/)
 
 ## Machine config preparation
 
 When generating the machine config for a node set the CNI to none.
 For example using a config patch:
 
+Create a `patch.yaml` file with the following contents:
+
+```yaml
+cluster:
+  network:
+    cni:
+      name: none
+```
+
 ```bash
 talosctl gen config \
     my-cluster https://mycluster.local:6443 \
-    --config-patch '[{"op":"add", "path": "/cluster/network/cni", "value": {"name": "none"}}]'
+    --config-patch @patch.yaml
 ```
 
 Or if you want to deploy Cilium in strict mode without kube-proxy, you also need to disable kube proxy:
 
+Create a `patch.yaml` file with the following contents:
+
+```yaml
+cluster:
+  network:
+    cni:
+      name: none
+  proxy:
+    disabled: true
+```
+
 ```bash
 talosctl gen config \
     my-cluster https://mycluster.local:6443 \
-    --config-patch '[{"op": "add", "path": "/cluster/proxy", "value": {"disabled": true}}, {"op":"add", "path": "/cluster/network/cni", "value": {"name": "none"}}]'
+    --config-patch @patch.yaml
 ```
 
-## Method 1: Helm install
+### Installation using Cilium CLI
+
+> Note: It is recommended to template the cilium manifest using helm and use it as part of Talos machine config, but if you want to install Cilium using the Cilium CLI, you can follow the steps below.
+
+Install the [Cilium CLI](https://docs.cilium.io/en/v1.13/gettingstarted/k8s-install-default/#install-the-cilium-cli) following the steps here.
+
+#### With kube-proxy
+
+```bash
+cilium install \
+    --helm-set=ipam.mode=kubernetes \
+    --helm-set=kubeProxyReplacement=disabled \
+    --helm-set=securityContext.capabilities.ciliumAgent="{CHOWN,KILL,NET_ADMIN,NET_RAW,IPC_LOCK,SYS_ADMIN,SYS_RESOURCE,DAC_OVERRIDE,FOWNER,SETGID,SETUID}" \
+    --helm-set=securityContext.capabilities.cleanCiliumState="{NET_ADMIN,SYS_ADMIN,SYS_RESOURCE}" \
+    --helm-set=cgroup.autoMount.enabled=false \
+    --helm-set=cgroup.hostRoot=/sys/fs/cgroup
+```
+
+#### Without kube-proxy
+
+```bash
+export KUBERNETES_API_SERVER_ADDRESS=<replace with api server endpoint here>
```

Comment:
just curious, does it need the public endpoint, or the 10.96.0.1 would work with kube-proxy?

Reply:
also, I'm always confused if that's the `endpoint` (like URL), or `endpoint` as the IP address/hostname (probably an example?)

Reply:
10.96.0.1, no since kube-proxy is not there :sweat_smile: